pkg/metamask/abi/parsing_test.go

```diff
@@ -40,7 +41,25 @@ func TestJsonAbi(t *testing.T) {
 	callData, err := parseNew(data)
 	require.NoError(t, err)
 
-	resJson, err := getJsonAbi(callData.Signature)
+	resJson, err := getJsonAbi(callData.Signature, callData.Inputs)
+	require.NoError(t, err)
+	require.Equal(t, string(resJson), expectedJson)
+}
+
+func TestJsonAbiPayments(t *testing.T) {
+	expectedJson := `[{"name":"transfer","type":"function","inputs":[{"type":"address"},{"type":"uint256"},{"type":"[(address, uint256)]"}]}]`
```

Comment:
Payments ABI-JSON representation isn't correct. See [this](https://github.com/ethereum/eth-abi/issues/35) for example.